### ISSUE TITLE
Why are the results of the three test cases I ran not very good?

### ISSUE THREAD

**Issue by fx-hit** [bytedance/TextHarmony#3]
After struggling with the environment setup (using A100 and PyTorch 2.1.0 cuda11.8), I was finally able to run it, but the results of the three test cases were not very good, with errors in the generated texts:

"Generate an image according to the caption. A cake of \"Good Time\". Outdoor, bright, sunny, happiness."
![0_0_0](https://github.com/user-attachments/assets/52839a24-0b20-48f9-9b37-f810f7c5dba1)

"Generate an image according to the caption. A T-shirt of \"Keep Focused\". Outdoor, bright, sunny, happiness, 24k."
![1_0_0](https://github.com/user-attachments/assets/60e2375c-9724-4a16-9695-dd180af674a8)

"Generate an image according to the caption. Photo of A book cover of \"Summer Love\". Outdoor, bright, sunny, happiness, 24k."
![2_0_0](https://github.com/user-attachments/assets/4af31585-51dc-4a17-a3a8-1b0848ae21b2)

**Comment by fx-hit** [comment]
The results in the paper were excellent:
![image](https://github.com/user-attachments/assets/6dfb3b3d-a5e0-45f9-ac4d-8290c24bf896)

**Comment by prefixRAINSTARsuffix** [comment]
@fx-hit Please try using torch2.0.1+cuda117

**Comment by fx-hit** [comment]
> @fx-hit Please try using torch2.0.1+cuda117

Switched to PyTorch 2.0.1 + CUDA 11.7, but there are still some errors with text generation.
![1_0_0 (2)](https://github.com/user-attachments/assets/acd1a157-f551-4e28-abdb-cd065ac34be8)

Package                       Version
----------------------------- ------------
absl-py                       1.4.0
accelerate                    0.31.0
addict                        2.4.0
adjustText                    1.1.1
aiofiles                      23.2.1
aiohttp                       3.9.3
aiosignal                     1.3.1
albumentations                1.3.1
aliyun-python-sdk-core        2.16.0
aliyun-python-sdk-kms         2.16.5
altair                        5.3.0
annotated-types               0.7.0
antlr4-python3-runtime        4.9.3
anyio                         4.4.0
async-timeout                 4.0.3
attr                          0.3.2
attrs                         23.2.0
braceexpand                   0.1.7
cachetools                    5.3.2
certifi                       2024.2.2
cffi                          1.17.1
charset-normalizer            3.3.2
click                         8.1.7
clip                          0.2.0
cmake                         3.28.1
colorama                      0.4.6
compel                        2.0.3
contourpy                     1.2.0
crcmod                        1.7
cryptography                  43.0.3
cycler                        0.12.1
datasets                      2.14.0
defusedxml                    0.7.1
diffusers                     0.20.0
dill                          0.3.7
dnspython                     2.6.1
docker-pycreds                0.4.0
easydict                      1.13
einops                        0.6.1
email_validator               2.1.1
exceptiongroup                1.2.1
fastapi                       0.111.0
fastapi-cli                   0.0.4
ffmpy                         0.3.2
filelock                      3.14.0
fire                          0.6.0
fonttools                     4.48.1
frozenlist                    1.4.1
fsspec                        2024.2.0
ftfy                          6.2.0
gitdb                         4.0.11
GitPython                     3.1.43
google-auth                   2.27.0
google-auth-oauthlib          1.2.0
gradio                        4.36.1
gradio_client                 1.0.1
gradio_image_prompter         0.1.0
grpcio                        1.60.1
h11                           0.14.0
hjson                         3.1.0
httpcore                      1.0.5
httptools                     0.6.1
httpx                         0.27.0
huggingface-hub               0.20.3
idna                          3.6
imageio                       2.33.1
importlib-metadata            7.0.1
importlib-resources           6.1.1
Jinja2                        3.1.3
jmespath                      0.10.0
joblib                        1.3.2
jsonschema                    4.22.0
jsonschema-specifications     2023.12.1
kiwisolver                    1.4.5
lazy_loader                   0.3
linkify-it-py                 2.0.3
lit                           17.0.6
lmdeploy                      0.5.0
Markdown                      3.5.2
markdown-it-py                2.2.0
MarkupSafe                    2.1.5
matplotlib                    3.8.4
mdit-py-plugins               0.3.3
mdurl                         0.1.2
mmcv                          2.2.0
mmengine                      0.10.5
mmengine-lite                 0.10.4
model-index                   0.1.11
mpmath                        1.3.0
multidict                     6.0.5
multiprocess                  0.70.15
MultiScaleDeformableAttention 1.0
mypy-extensions               1.0.0
networkx                      3.2.1
ninja                         1.11.1
nltk                          3.8.1
numpy                         1.26.4
nvidia-cublas-cu11            11.11.3.6
nvidia-cuda-cupti-cu11        11.8.87
nvidia-cuda-nvrtc-cu11        11.8.89
nvidia-cuda-runtime-cu11      11.8.89
nvidia-cudnn-cu11             8.7.0.84
nvidia-cufft-cu11             10.9.0.58
nvidia-curand-cu11            10.3.0.86
nvidia-cusolver-cu11          11.4.1.48
nvidia-cusparse-cu11          11.7.5.86
nvidia-nccl-cu11              2.19.3
nvidia-nvtx-cu11              11.8.86
oauthlib                      3.2.2
omegaconf                     2.3.0
openai-clip                   1.0.1
opencv-python                 4.9.0.80
opencv-python-headless        4.9.0.80
opendatalab                   0.0.10
openmim                       0.3.9
openxlab                      0.1.2
ordered-set                   4.1.0
orjson                        3.10.3
oss2                          2.17.0
packaging                     24.1
palettable                    3.3.3
pandas                        2.2.0
peft                          0.11.1
Pillow                        10.0.0
pip                           24.0
platformdirs                  4.2.0
protobuf                      4.23.4
psutil                        5.9.8
py-cpuinfo                    9.0.0
pyarrow                       12.0.1
pyasn1                        0.5.1
pyasn1-modules                0.3.0
pycocoevalcap                 1.2
pycocotools                   2.0.6
pycparser                     2.22
pycryptodome                  3.21.0
pydantic                      1.10.14
pydantic_core                 2.18.4
pydub                         0.25.1
Pygments                      2.18.0
pynvml                        11.5.0
pyparsing                     3.1.1
pyre-extensions               0.0.29
python-dateutil               2.8.2
python-dotenv                 1.0.1
python-multipart              0.0.9
pytz                          2023.4
PyYAML                        6.0.1
qudida                        0.0.4
referencing                   0.35.1
regex                         2023.12.25
requests                      2.28.2
requests-oauthlib             1.3.1
rich                          13.4.2
rpds-py                       0.18.1
rsa                           4.9
ruff                          0.4.8
safetensors                   0.4.2
scikit-image                  0.22.0
scikit-learn                  1.3.1
scipy                         1.11.1
semantic-version              2.10.0
sentencepiece                 0.1.99
sentry-sdk                    1.43.0
setproctitle                  1.3.3
setuptools                    60.2.0
shellingham                   1.5.4
shortuuid                     1.0.13
six                           1.16.0
smmap                         5.0.1
sniffio                       1.3.1
starlette                     0.37.2
supervision                   0.21.0
sympy                         1.12
tabulate                      0.9.0
tensorboard                   2.17.0
tensorboard-data-server       0.7.2
tensorboardX                  2.6.2.2
termcolor                     2.4.0
threadpoolctl                 3.2.0
tifffile                      2024.1.30
tiktoken                      0.7.0
timm                          1.0.3
tokenizers                    0.13.3
tomli                         2.0.1
tomlkit                       0.12.0
toolz                         0.12.1
torch                         2.0.1+cu117
torchaudio                    2.0.2+cu117
torchvision                   0.15.2+cu117
tqdm                          4.65.2
transformers                  4.31.0
triton                        2.0.0
typer                         0.12.3
typing_extensions             4.9.0
typing-inspect                0.9.0
tzdata                        2023.4
uc-micro-py                   1.0.3
ujson                         5.10.0
urllib3                       1.26.20
uvicorn                       0.30.1
uvloop                        0.19.0
wandb                         0.17.2
watchfiles                    0.22.0
wcwidth                       0.2.13
webdataset                    0.2.48
websockets                    11.0.3
Werkzeug                      3.0.1
wheel                         0.43.0
xformers                      0.0.20
xxhash                        3.4.1
yapf                          0.40.2
yarl                          1.9.4
zipp                          3.17.0

**Comment by prefixRAINSTARsuffix** [comment]
@fx-hit  It is weird. Maybe try aligning all package versions (for example mmcv-full==1.7.0 and flash-attn==2.0.4). The results generated in our local environment are correct.

**Comment by fx-hit** [comment]
@fx-hit It is weird. Maybe try aligning all package versions (for example mmcv-full==1.7.0 and flash-attn==2.0.4). The results generated in our local environment are correct.

I have checked the currently open-source code. The mmcv is only used in utils/parse_args.py, where its function is to parse configuration items, and the impact is minimal. The flash-attn is also only triggered during the training process; commenting out the import flash_attn part does not affect inference.

**Comment by zhangzjn** [comment]
Following the README I got the better results, but slight artifact remains. 
![0_0_0](https://github.com/user-attachments/assets/fb61c4ac-6a99-4314-8141-ec3479faafb7)
![1_0_0](https://github.com/user-attachments/assets/802f7744-68cc-4f9e-a3e2-8785638677c4)
![2_0_0](https://github.com/user-attachments/assets/48b1fe59-2b79-4074-a78b-38a40229ab9a)

**Comment by fx-hit** [comment]
> Following the README I got the better results, but slight artifact remains. ![0_0_0](https://private-user-images.githubusercontent.com/26211183/390012838-fb61c4ac-6a99-4314-8141-ec3479faafb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NjEyODksIm5iZiI6MTczMjc2MDk4OSwicGF0aCI6Ii8yNjIxMTE4My8zOTAwMTI4MzgtZmI2MWM0YWMtNmE5OS00MzE0LTgxNDEtZWMzNDc5ZmFhZmI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDAyMjk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiYWFmNzZjNTAyYTE5YjdhYmEwM2JlNTVmNTAwOWMyOTg0ZTY5ZmNlOTRhNWFiMDFmYWJiYWJmODQ3YzZkNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zvzuViatet02rLt-ZWGicLsyZEmZ2-nBDUGrgWGIeJA) ![1_0_0](https://private-user-images.githubusercontent.com/26211183/390012923-802f7744-68cc-4f9e-a3e2-8785638677c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NjEyODksIm5iZiI6MTczMjc2MDk4OSwicGF0aCI6Ii8yNjIxMTE4My8zOTAwMTI5MjMtODAyZjc3NDQtNjhjYy00ZjllLWEzZTItODc4NTYzODY3N2M0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDAyMjk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUyM2RjMmUyMmI2MGI5NWUxNGMyMTQyMWZkMjZkMjZlN2IwZDllN2Q0ODY1NjEyNDllMTVmOWYzZGMxMDFhMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WN9tsx9KFz_WzivtedUeM7RKs3F1n-mRJDCGjIWFBBo) ![2_0_0](https://private-user-images.githubusercontent.com/26211183/390012950-48b1fe59-2b79-4074-a78b-38a40229ab9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NjEyODksIm5iZiI6MTczMjc2MDk4OSwicGF0aCI6Ii8yNjIxMTE4My8zOTAwMTI5NTAtNDhiMWZlNTktMmI3OS00MDc0LWE3OGItMzhhNDAyMjlhYjlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDAyMjk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzNmY2ZjFjMWRiOTQ4OGZmNjY2ZTIyYTliMzZjYjU2ZjZjNDZkZjNmYWNiZTdkZjEzMDdkNmMxNTI2MTIzNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4cQZpa915L0SV1Wff3fZuq9ftRlN5H-uddV_QsLNmpY)

Is it necessary for all package versions to be consistent? It seems a bit strange. Generally speaking, if there's an issue with the environment, either the code won't run at all, or the results produced will be completely wrong.
